mpathconfig/configresource.go

```diff
@@ -237,15 +237,21 @@ func addOption(section *Section, option string) {
 	}
 }
 
-func isSection(line string) bool {
-	line = strings.TrimSpace(line)
+//checks for the section
+func isSection(line string) (bool, error) {
+	line = strings.TrimSpace(line)	
 	prefixes := []string{"defaults", "blacklist", "blacklist_exceptions", "devices", "device", "multipaths", "multipath"}
-	for _, prefix := range prefixes {
-		if strings.HasPrefix(line, prefix) {
-			return true
+        for _, prefix := range prefixes {
+		r, err := regexp.Compile("^"+prefix+"\\s*{")	
```

Comment:
@skshetti i think we need to just make sure these keys are whole words. here are the possibilities for eg:

multipaths {
multipath
{
So, if we change to ```r, err := regexp.Compile("^"+prefix+"\\s+``` i think that should do.

Reply:
If we look for the keys as whole words what if there is something like this:
defaults {
             multipath         
}
i.e if there is a parameter multipath and no value..there isn't any check for the parameters without any values or wrong values except for the device section where the vendor is nimble.
So in this case "nimbletune --set  category multipath" would return error.

thats why i first modified the multipath.conf file by removing newlines between section name and { so that  "regexp.Compile("^"+prefix+"\\s*{")" would work fine in any case.

Reply:
no, param names cannot be same as section names(they will never be). So if you compare whole worlds, thats good enough to parse sections. The customer issue was with param multipath_dir, where we were looking for multipath section.

Reply:
so param names shouldn't be similar to section names but there is no check being made for these parameters right? And the user can modify /etc/multipath.conf right? so what if he does this.?

Reply:
multipath checks for them. If unknown params are added to multipath.conf, it will throw an error. you can try after adding them and run "multipath -ll". So, i think users adding unknown params is very remote. Just to minimize churn in this code(as its used by multiple plugins), we can handle this particular customer issue.

Reply:
Okay...so no need to remove the newlines between section name and { and just checking for the whole word would be enough right?

Reply:
Yes, i think that should resolve the issue. Please try to test it out with various combinations(valid keys) and verify as well. Thanks!